### PR TITLE
Pyic 3408 contact us links

### PR DESCRIPTION
**Pull request by DanCorderIPV**
DO NOT MERGE - until new contact page is actually live

### What changed

Contact links updated to new format as per: https://docs.google.com/document/d/1igEUlA8gwOsv4YD4xtwA42CeIXaR2Yv9gVnQEH3N7BY/edit?pli=1

### Why did it change

There is a new contact page being put live around the end of October. When that happens we need to update our links.

### Issue tracking

- [PYIC-3408](https://govukverify.atlassian.net/browse/PYIC-3408)

[PYIC-3408]: https://govukverify.atlassian.net/browse/PYIC-3408?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/app/ipv/middleware.js [hunk 1]
```diff
@@ -423,7 +419,7 @@ module.exports = {
   },
   allTemplates: async (req, res, next) => {
     try {
-      const directoryPath = "/app/src/views/ipv";
+      const directoryPath = __dirname + "/../../views/ipv";
```
Comment (DanCorderIPV) [inline src/app/ipv/middleware.js:422]: I have tested this locally and on a dev stack and it still works

File: src/locales/cy/translation.json [hunk 2]
```diff
@@ -15,13 +15,12 @@
...
       "phaseBanner": {
         "tag": "beta",
-        "content": "Mae hwn yn wasanaeth newydd – bydd eich <a class=\"govuk-link\" rel=\"noopener\" target=\"_blank\" href=\"https://signin.account.gov.uk/contact-us\">adborth (agor mewn tab newydd)</a> yn ein helpu i’w wella."
+        "content": "Mae hwn yn wasanaeth newydd – bydd eich <a class=\"govuk-link\" rel=\"noopener\" target=\"_blank\" href=\"{{url}}\">adborth (agor mewn tab newydd)</a> yn ein helpu i’w wella."
       }
     },
     "shared": {
       "govUKHomepageButtonHref": "https://www.gov.uk/",
       "govUKHomepageButtonText": "Ewch i hafan GOV.UK",
-      "contactLinkHref": "https://signin.account.gov.uk/contact-us",
```
Comment (DanCorderIPV) [inline src/locales/cy/translation.json:24]: This is the same in English and Welsh so it doesn't need to be in the translation file

File: src/locales/cy/translation.json [hunk 3]
```diff
@@ -328,8 +327,7 @@
         "subHeading4": "Os ydych angen help",
         "paragraph8": "Cysylltwch â ni os:",
         "bullet3": "ydych angen help i orffen profi eich hunaniaeth gyda Swyddfa’r Post",
-        "bullet4": "hoffech beidio â mynd i Swyddfa’r Post ac eisiau profi eich hunaniaeth mewn ffordd arall",
-        "contactLinkHrefF2fCustom": "https://signin.account.gov.uk/contact-us-questions?theme=proving_identity&referer="
```
Comment (DanCorderIPV) [inline src/locales/cy/translation.json:332]: This isn't the same in English and Welsh but it should be, so it doesn't need to be in the translation file

File: src/views/errors/service-unavailable-s3/service-unavailable.html [hunk 4]
```diff
@@ -116,8 +116,8 @@ <h2 class="govuk-visually-hidden">Support links</h2>
             </li>
 
             <li class="govuk-footer__inline-list-item">
-              <a class="govuk-footer__link" href="https://signin.account.gov.uk/contact-us">
-                Support
+              <a class="govuk-footer__link" target="_blank" rel="noopener noreferrer" href="https://home.account.gov.uk/contact-gov-uk-one-login?fromURL=https://signin.account.gov.uk/service-unavailable">
```
Comment (DanCorderIPV) [inline src/views/errors/service-unavailable-s3/service-unavailable.html:119]: Make this link consistent with all the others

File: src/app.js [hunk 5]
```diff
@@ -128,6 +128,15 @@ app.use((req, res, next) => {
   }
 });
 
+// Calculate and store the contact us URL so that every page template can access it.
+app.use((req, res, next) => {
+  const currentPageUrl = "https://signin.account.gov.uk" + req.originalUrl;
```
Comment (blakeyp) [inline src/app.js]: Do we need to provide the prefix or can we not get this from the request?

File: src/app.js [hunk 6]
```diff
@@ -128,6 +128,15 @@ app.use((req, res, next) => {
...
+  res.locals.contactUsUrl =
+    "https://home.account.gov.uk/contact-gov-uk-one-login?fromURL=" +
```
Comment (blakeyp) [inline src/app.js]: Could we store this url at least in the config file and maybe pull this middleware out to its own function/file so it's easier to find?

File: src/lib/config.js [hunk 7]
```diff
@@ -4,6 +4,13 @@ const cfenv = require("cfenv");
 const appEnv = cfenv.getAppEnv();
 const serviceConfig = {};
 
+const DEVELOPMENT_ENVIRONMENT_NAME = "development";
+const LOCAL_ENVIRONMENT_NAME = "local";
+// We need to limit this dev/debug page to development environments
+const ENABLE_ALL_TEMPLATES_PAGE =
+  process.env.NODE_ENV === DEVELOPMENT_ENVIRONMENT_NAME ||
```
Comment (blakeyp) [inline src/lib/config.js]: Is NODE_ENV set to development in build? 🤔

File: src/views/errors/service-unavailable-s3/service-unavailable.html (hunk 4 above)
Comment (blakeyp) [reply src/views/errors/service-unavailable-s3/service-unavailable.html:119]: Todo

File: src/lib/config.js (hunk 7 above)
Comment (shivanshuit914) [reply src/lib/config.js]: Yes, I can see `nodeEnv: "development"` in template.yaml file.

File: src/app.js (hunk 5 above)
Comment (shivanshuit914) [reply src/app.js]: We can get it from the req. I will update it. Just to confirm signin url will be diff depending on the env.

Comment (blakeyp) [reply src/app.js]: Yep it should be per env

File: src/views/errors/service-unavailable-s3/service-unavailable.html (hunk 4 above)
Comment (shivanshuit914) [reply src/views/errors/service-unavailable-s3/service-unavailable.html:119]: I am not sure how we can pass the dynamic contact url in static html page.

File: deploy/template.yaml [hunk 8]
```diff
@@ -63,6 +63,7 @@ Mappings:
       desiredTaskCount: 1
       assetsDomain: ""
       assetsPath: ""
+      contactUrl: "https://home.account.gov.uk/contact-gov-uk-one-login"
```
Comment (blakeyp) [inline deploy/template.yaml]: *home.build.account.gov.uk (we can point to build from dev)

File: deploy/template.yaml [hunk 9]
```diff
@@ -76,6 +77,7 @@ Mappings:
...
```
Comment (blakeyp) [inline deploy/template.yaml]: *home.build.account.gov.uk

File: deploy/template.yaml [hunk 10]
```diff
@@ -93,6 +95,7 @@ Mappings:
       desiredTaskCount: 2
       assetsDomain: "https://assets.build.account.gov.uk"
       assetsPath: "https://assets.build.account.gov.uk/core-front"
+      contactUrl: "https://home.account.gov.uk/contact-gov-uk-one-login"
```
Comment (blakeyp) [inline deploy/template.yaml]: *home.build.account.gov.uk

File: deploy/template.yaml [hunk 11]
```diff
@@ -106,6 +109,7 @@ Mappings:
       desiredTaskCount: 1
       assetsDomain: "https://assets.staging.account.gov.uk"
       assetsPath: "https://assets.staging.account.gov.uk/core-front"
+      contactUrl: "https://home.account.gov.uk/contact-gov-uk-one-login"
```
Comment (blakeyp) [inline deploy/template.yaml]: *home.staging.account.gov.uk

File: deploy/template.yaml [hunk 12]
```diff
@@ -119,6 +123,7 @@ Mappings:
       desiredTaskCount: 1
       assetsDomain: "https://assets.integration.account.gov.uk"
       assetsPath: "https://assets.integration.account.gov.uk/core-front"
+      contactUrl: "https://home.account.gov.uk/contact-gov-uk-one-login"
```
Comment (blakeyp) [inline deploy/template.yaml]: *home.integration.account.gov.uk

File: src/views/errors/service-unavailable-s3/service-unavailable.html (hunk 4 above)
Comment (blakeyp) [reply src/views/errors/service-unavailable-s3/service-unavailable.html:119]: Ahh didn't realise it's static html, we'll have to leave this one then

**Review by blakeyp — DISMISSED** [review]

**Review by blakeyp — APPROVED** [review]

**Pull request merged**